data/tutorials/language/1ds_01_arrays.md

```diff
@@ -0,0 +1,156 @@
+
+Arrays are commonly used in OCaml for tasks such as:
+
+- Storing and processing large amounts of data
+- Implementing algorithms that require random access and modification of elements
+- Working with matrices and other multi-dimensional data structures
+
+## Creating Arrays
+
+To create an array in OCaml, you can use the `[| ...; ... |]` syntax, which allows you to specify the values of each element directly. For example, to create an array with the values 1, 2, 3, 4, and 5, you will write `[| 1; 2; 3; 4; 5 |]`:
+
+```ocaml
+# [| 1; 2; 3; 4; 5 |];;
+- : int array = [|1; 2; 3; 4; 5|]
+```
+
+Alternatively, you can create an array using the `Array.make` function, which takes two arguments: the length of the array and the initial value of each element. For example, to create an array of length 5 with all elements initialized to 0, you can write:
+
+```ocaml
+# let zeroes = Array.make 5 0;;
+val zeroes : int array = [|0; 0; 0; 0; 0|]
+```
+
+`Array.init` generates an array of a given length by applying a function to each index of the array, starting at 0. The following line of code creates an array containing the first 5 even numbers using a function which doubles its argument:
+
+```ocaml
+# let even_numbers = Array.init 5 (fun i -> i * 2);;
+val even_numbers : int array = [|0; 2; 4; 6; 8|]
+```
+
+## Accessing Array Elements
+
+You can access individual elements of an array using the `.(index)` syntax, with the index of the element you want to access. The index of the first element is 0, and the index of the last element is one less than the size of the array.  For example, to access the third element of an array `even_numbers`, you would write:
+
+```ocaml
+# even_numbers.(2);;
+- : int = 4
+```
+
+## Modifying Array Elements
+
+To modify an element in an array, we simply assign a new value to it using the indexing operator. For example, to change the value of the third element of the array `even_numbers` created above to 42, we have to write:
+
+```ocaml
+# even_numbers.(2) <- 42;;
+- : unit = ()
+```
+
+Note that this modification doesn’t return the modified array: the value returned by this operation is `unit`, `even_numbers` is modified in place in the current scope. Updating an array's content is a side effect.
+
+## The Standard Library `Array` Module
+
+OCaml provides several useful functions for working with arrays. Here are some of the most common ones:
+
+### Length of an Array
+
+The `Array.length` function returns the size of an array:
+
+```ocaml
+# Array.length even_numbers;;
+- : int = 5
+```
+
+### Iterate on an Array
+
+`Array.iter` applies a function to each element of an array, one at a time. The given function must return `unit`, operating by side effect. To print all the elements of the array `zeroes` created above, we can apply `print_int` to each element:
+
+```ocaml
+# Array.iter (fun x -> print_int x; print_string " ") zeroes;;
+0 0 0 0 0 - : unit = ()
+```
+
```

Comment:
I think this could be a good place to comment, that `for` loops tend to work well for iterating on arrays too.
One option would be to translate the `iter` example:
```ocaml
# for i = 0 to Array.length a - 1 do print _int zeroes.(i); print_string " " done;;
 ```
 which offer the occasion to point that the control flow of the loop is more complex : what happens if we use the wrong start index, or forgot the `-1`? Of if we access another array ?
 Another option would be the Fisher–Yates shuffle
 ```ocaml
 let exchange a i j = let tmp = a.(i) in a.(i) <- a.(j); a.(j) <- tmp
 let shuffle a = for i = Array.length a - 1 downto 1 do
    exchange i (Random.int (i+1))
 done
 ```
 which gives a more consistent example of which kind of array code ends up using loops.